### PR TITLE
MES-4922 Fixed showing aborted attempt in speed requirement

### PR DESCRIPTION
**Pull request by edwardwilson**
## Description
Fixed the code checks that broke when the data changed from string to numbers

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/modules/tests/test-data/cat-a-mod1/avoidance/avoidance.selector.ts [hunk 1]
```diff
@@ -3,6 +3,6 @@ import { TestData, Avoidance } from '@dvsa/mes-test-schema/categories/AM1';
 export const getAvoidance = (testData: TestData): Avoidance => testData.avoidance;
 
 export const getAvoidanceAttempted = (avoidance: Avoidance) : boolean => {
-  if (avoidance.firstAttempt || avoidance.secondAttempt) return true;
+  if (avoidance.firstAttempt !== undefined || avoidance.secondAttempt !== undefined) return true;
```
Comment (szabikr) [inline src/modules/tests/test-data/cat-a-mod1/avoidance/avoidance.selector.ts]: you can return the expression evaluation and remove the if statement

File: src/pages/debrief/cat-a-mod1/components/speed-check-debrief-card/speed-check-debrief-card.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<ion-card id="speed-check" *ngIf="emergencyStop.firstAttempt || avoidance.firstAttempt">
+<ion-card id="speed-check" *ngIf="emergencyStop.firstAttempt !== undefined || avoidance.firstAttempt !== undefined">
```
Comment (szabikr) [inline src/pages/debrief/cat-a-mod1/components/speed-check-debrief-card/speed-check-debrief-card.html:1]: use the same function that is implemented in the selector

**Review by szabikr — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
Agree with Szabi's comments. 
Looks good.

File: src/pages/debrief/cat-a-mod1/components/speed-check-debrief-card/speed-check-debrief-card.html (hunk 2 above)
Comment (edwardwilson) [reply src/pages/debrief/cat-a-mod1/components/speed-check-debrief-card/speed-check-debrief-card.html:1]: The selector checks if the avoidance was attempted, it doesn't check if the emergency stop was like above. There could be a function created if desired but the suggestion to use the selector wouldn't work.

File: src/modules/tests/test-data/cat-a-mod1/avoidance/avoidance.selector.ts (hunk 1 above)
Comment (edwardwilson) [reply src/modules/tests/test-data/cat-a-mod1/avoidance/avoidance.selector.ts]: Yes correct! We can now before there was a typical issue with them typed as strings but compared to undefiend we can use a lamba statment

**Pull request merged**